### PR TITLE
Added docs on the design and architecture

### PR DESCRIPTION
**Pull request by Amber1990Zhang**
Design and architecture of Nebula Graph, written based on pr **#591 and Nebula Graph Internal** so it is duplicate with pr #591 in some contents, but I change the organization method in this pr to make it more clear and simple.

### CODE REVIEW

File: docs/manual-EN/overview/design-and-arc.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+This document is to walk you through on how **Nebula Graph** is designed and why, and it will be separated into two parts: Industry Overview and Nebula Graph Architecture.
+
+## Industry Overview
+
+### OLAP & OLTP in Graph 
+
+![meetup1-04](https://user-images.githubusercontent.com/42762957/64223265-dc573680-cf05-11e9-98bc-93de4fd32463.jpg)
+
+The axis in the above picture shows the different requirement for query latency. Like traditional database, graph database can be divided into two parts: OLAP and OLTP. OLAP cares more about offline analysis while OLTP prefers online processing. Graph computing framework in OLAP is used to analysis data based on graph structure. And it's similar to OLAP in traditional database. But it has features which are not available in traditional database, one is iterative algorithm based on graph. A typical example is the PageRank algorithm from Google, which obtains the relevance of web pages through constant iterative computing. Another example is the commonly-used LPA algorithm.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: requirement  => requirements

File: docs/manual-EN/overview/design-and-arc.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+Along the axis to right, there comes the graph streaming field, which is the combination of basic computing and streaming computing. A relational network is not a static structure, rather, it constantly changes in the business: be it graph structure or graph properties. Computing in this filed 
+is often triggered by events and its latency is in second.
+
+Right beside the graph streaming is the online response system, whose requirement for latency is extremely high, which should be in millisecond. 
+The right most field is graph database and its use cases are completely different from the one on the left. The **Nebula Graph** we're working on can find its usage here in OLTP.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: rightmost?

File: docs/manual-EN/overview/design-and-arc.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+## Nebula Graph Architecture
+
+Designed based on the above features, **Nebula Graph** is an open source, distributed, lightning-fast graph database, it is composed of four components: storage service, meta service, query engine and client. 
+
+![meetup1-13](https://user-images.githubusercontent.com/42762957/64231577-9c527c80-cf22-11e9-9044-9a739a22c42a.jpg)
+
+The dash line in the above picture divided computing and storage as two independent parts, the upper is the computing service, each machine or virtual machine is stateless and never talks to other so it's easy to scale in or out; the lower is the storage service, it's stateful since data is stored there. Storage service can turn graph semantics into key-values and pass them to the KV-store below it. Between the two is the Raft protocol.
+
+The right side is the meta service, similar with the NameNode in HDFS, it stores all meta data like schema and controls scaling.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: similar to?

File: docs/manual-EN/overview/design-and-arc.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+The dash line in the above picture divided computing and storage as two independent parts, the upper is the computing service, each machine or virtual machine is stateless and never talks to other so it's easy to scale in or out; the lower is the storage service, it's stateful since data is stored there. Storage service can turn graph semantics into key-values and pass them to the KV-store below it. Between the two is the Raft protocol.
+
+The right side is the meta service, similar with the NameNode in HDFS, it stores all meta data like schema and controls scaling. 
+
+### Design thinking: Storage Service
+
+**Nebula Graph** adopted the **share-nothing** distributed architecture in storage so nodes do not share memory or storage, which means there is no central nodes in the whole system. Benefits of such design are: 
+
+- Easy to scale
+- The overall system continues operating despite individual crash
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: individual crash => an individual crash

File: docs/manual-EN/overview/design-and-arc.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+Another design is the **separation of computing and storage**, and the benefits are as follows:
+
+- Scalability. Separating storage from computing makes storage service flexible,thus it's easy to scale out or in.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: space between the comma and 'thus'

File: docs/manual-EN/overview/design-and-arc.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+Migrating the partitions on an overworked server to other relatively idle servers to increases availability and capacity of the system.
+
+### Design-thinking: Meta Service
+
+The binary of the meta service is **nebula-metad**. Here is the list of its main functionalities:
+
+-  User management
+
+   In **Nebula Graph** different roles are assigned diverse privileges. We provide the following native roles: Global Admin, Graph Space Admin, User,Guest.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: User and Guest

File: docs/manual-EN/overview/design-and-arc.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+- Cluster configuration management
+
+   Meta service manages the servers and partitions in the cluster, e.g. records location of the partitions, receives heartbeat from servers, etc. It balances the partitions and manages the communication traffic in case of server failure.
+- Graph space management
+
+   **Nebula Graph** supports multiple graph spaces. Data in different graph spaces are physically isolated. Meta service stores the metadata of all spaces in the cluster and tracks changes that take place in these spaces, like adding, dropping space, modifying graph space configuration (Raft copies).
+-   Schema management
+
+    **Nebula Graph** is a strong typed database.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: a strong typed database?

File: docs/manual-EN/overview/design-and-arc.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+The binary of the query engine is **nebula-graphd**. Each nebula-graphd instance is stateless and never talks to other nebula-graphd. nebula-graphd only talks to the storage service and the meta service. That makes it trivial to expand or shrink the query engine cluster.
+
+ The query engine accepts the message from the client and generates the execution plan after the lexical parsing (Lexer), semantic analysis (Parser) and the query optimization. Then the execution plan will be passed to the execution engine. The query execution engine takes the query plans and interacts with meta server and the storage engine to retrieve the schema and data.
+
+ The main optimizations of the query engine are:
+
+ - Asynchronous and parallel execution
+
+    I/O operations and network transmission are time-consuming. Thus asynchronous and parallel operations are widely adopted in the query engine to reduce the latency and to improve the overall throughput.Also, a separate resource pool is set for each query to avoid the long tail effect of those time-consuming queries.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: space before ''Also'

File: docs/manual-EN/overview/design-and-arc.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+ - Pushing down computation
+
+    In a distributed system, transferring large amount of data on the network really hurts the overall latency. In **Nebula Graph**, the query engine will make decisions to push some filter and aggregation down to the storage service. The purpose is to reduce the amount of data passing back from the storage.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: "really hurts the overall latency" a little bit weird

Comment (Amber1990Zhang) [reply docs/manual-EN/overview/design-and-arc.md]: How about hurts => extend?

File: docs/manual-EN/overview/design-and-arc.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+This document is to walk you through on how **Nebula Graph** is designed and why, and it will be separated into two parts: Industry Overview and Nebula Graph Architecture.
+
+## Industry Overview
+
+### OLAP & OLTP in Graph 
+
+![meetup1-04](https://user-images.githubusercontent.com/42762957/64223265-dc573680-cf05-11e9-98bc-93de4fd32463.jpg)
+
+The axis in the above picture shows the different requirements for query latency. Like traditional database, graph database can be divided into two parts: OLAP and OLTP. OLAP cares more about offline analysis while OLTP prefers online processing. Graph computing framework in OLAP is used to analyse data based on graph structure. And it's similar to OLAP in traditional database. But it has features which are not available in traditional database, one is iterative algorithm based on graph. A typical example is the PageRank algorithm from Google, which obtains the relevance of web pages through constant iterative computing. Another example is the commonly-used LPA algorithm.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: Like a traditional database

File: docs/manual-EN/overview/design-and-arc.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+Here **V** is a set of nodes, aka vertices, **E** is a set of directional edges, **P<sub>V</sub>** represents properties on vertices, and **P<sub>E</sub>** is the properties on edges.
+
+## Nebula Graph Architecture
+
+Designed based on the above features, **Nebula Graph** is an open source, distributed, lightning-fast graph database, it is composed of four components: storage service, meta service, query engine and client. 
+
+![meetup1-13](https://user-images.githubusercontent.com/42762957/64231577-9c527c80-cf22-11e9-9044-9a739a22c42a.jpg)
+
+The dash line in the above picture divided computing and storage as two independent parts, the upper is the computing service, each machine or virtual machine is stateless and never talks to other so it's easy to scale in or out; the lower is the storage service, it's stateful since data is stored there. Storage service can turn graph semantics into key-values and pass them to the KV-store below it. Between the two is the Raft protocol.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: dashed line

File: docs/manual-EN/overview/design-and-arc.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+The right side is the meta service, similar to the NameNode in HDFS, it stores all meta data like schema and controls scaling.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: metadata

File: docs/manual-EN/overview/design-and-arc.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+![meetup1-13](https://user-images.githubusercontent.com/42762957/64231577-9c527c80-cf22-11e9-9044-9a739a22c42a.jpg)
+
+The dash line in the above picture divided computing and storage as two independent parts, the upper is the computing service, each machine or virtual machine is stateless and never talks to other so it's easy to scale in or out; the lower is the storage service, it's stateful since data is stored there. Storage service can turn graph semantics into key-values and pass them to the KV-store below it. Between the two is the Raft protocol.
+
+The right side is the meta service, similar to the NameNode in HDFS, it stores all meta data like schema and controls scaling. 
+
+### Design thinking: Storage Service
+
+**Nebula Graph** adopted the **share-nothing** distributed architecture in storage so nodes do not share memory or storage, which means there is no central nodes in the whole system. Benefits of such design are:
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: there is no => there are no

File: docs/manual-EN/overview/design-and-arc.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+- Easy to scale
+- The overall system continues operating despite individual crash
+
+Another design is the **separation of computing and storage**, and the benefits are as follows:
+
+- Scalability. Separating storage from computing makes storage service flexible, thus it's easy to scale out or in.
+- Availability. Recovery from vertex failure can be performed quickly. 
+
+The binary of storage service is **nebula-storaged**, which provides a key-value store. Multiple storage engines like RocksDB and HBase are supported, with RocksDB set as default engine. To build a resilient distributed system, [Raft](https://raft.github.io/) is implemented as the consensus algorithm.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: default engine =>  the default engine

File: docs/manual-EN/overview/design-and-arc.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+
+ The main optimizations of the query engine are:
+
+ - Asynchronous and parallel execution
+
+    I/O operations and network transmission are time-consuming. Thus asynchronous and parallel operations are widely adopted in the query engine to reduce the latency and to improve the overall throughput. Also, a separate resource pool is set for each query to avoid the long-tail effect of those time-consuming queries.
+
+ - Pushing down computation
+
+    In a distributed system, transferring large amount of data on the network really extends the overall latency. In **Nebula Graph**, the query engine will make decisions to push some filter and aggregation down to the storage service. The purpose is to reduce the amount of data passing back from the storage.
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: 'large amounts of' or 'a large amount of'

Comment (Amber1990Zhang) [reply docs/manual-EN/overview/design-and-arc.md]: a large amount of'

File: docs/manual-EN/overview/design-and-arc.md [hunk 16]
```diff
@@ -109,7 +109,7 @@ The binary of the query engine is **nebula-graphd**. Each nebula-graphd instance
 
  - Pushing down computation
 
-    In a distributed system, transferring large amount of data on the network really extends the overall latency. In **Nebula Graph**, the query engine will make decisions to push some filter and aggregation down to the storage service. The purpose is to reduce the amount of data passing back from the storage.
+    In a distributed system, transferring a large amount of data on the network really extends the overall latency. In **Nebula Graph**, the query engine will make decisions to push some filter and aggregation down to the storage service. The purpose is to reduce the amount of data passing back from the storage.
 
 ### Design-thinking: API and SDK
```
Comment (whitewum) [inline docs/manual-EN/overview/design-and-arc.md]: web SDK -> web console

File: docs/manual-EN/overview/design-and-arc.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+
+![meetup1-13](https://user-images.githubusercontent.com/42762957/64231577-9c527c80-cf22-11e9-9044-9a739a22c42a.jpg)
+
+The dashed line in the above picture divided computing and storage as two independent parts, the upper is the computing service, each machine or virtual machine is stateless and never talks to other so it's easy to scale in or out; the lower is the storage service, it's stateful since data is stored there. Storage service can turn graph semantics into key-values and pass them to the KV-store below it. Between the two is the Raft protocol.
+
+The right side is the meta service, similar to the NameNode in HDFS, it stores all metadata like schema and controls scaling. 
+
+### Design thinking: Storage Service
+
+**Nebula Graph** adopted the **share-nothing** distributed architecture in storage so nodes do not share memory or storage, which means there are no central nodes in the whole system. Benefits of such design are:
```
Comment (jude-zhu) [inline docs/manual-EN/overview/design-and-arc.md]: share-nothing or shared-nothing?

**Comment by jude-zhu** [comment]
Jenkins go

**Pull request merged**